apps/labs/posts/scipy-array-api.md

```diff
@@ -0,0 +1,441 @@
+
+-     assert_allclose(fft_res, fftpack_res)
++     xp_assert_close(fft_res, fftpack_res)
+```
+
+## Machinery, Helpers and Utilities
+
+I'll talk a bit about implementation details now - feel free to skip this section if you are just here for
+an overview!
+
+When we call `array_namespace`, a little more is done than just returning the namespace.
+If the environment variable `SCIPY_ARRAY_API` is not set, `array_api_compat.numpy`
+(the compatible wrapper for `numpy`) is returned, regardless of the type of the input array.
+This allows us to retain all pre-array API behaviour unless the user has set the environment variable.
+If it isn't set, we raise exceptions for known-bad subclasses
+(as set out in [the RFC](https://github.com/scipy/scipy/issues/18286)),
+like NumPy `MaskedArray`s and object arrays.
+After that, we retrieve the namespace using `array_api_compat.array_namespace`.
+
+The current testing infrastructure allows us to test with PyTorch (CPU & GPU) and CuPy through the command line.
+For example, `python dev.py test -b pytorch` tests with PyTorch, and `python dev.py test -b all` tests with
+NumPy, CuPy, PyTorch and `numpy.array_api`.
+[`numpy.array_api`](https://numpy.org/devdocs/reference/array_api.html) is a strict minimal implementation of the
+standard, which allows us to make sure that our array-agnostic code is _really_ array-agnostic -
+if our tests pass for `numpy.array_api`, then they will pass for any fully compliant library.
+PyTorch GPU can be enabled by setting the environment variable `SCIPY_DEVICE=cuda`.
+
+---
+
+## Status
+
+At the time of writing, SciPy has two of its submodules converted (in the sense I have described above):
+`cluster` and `fft`.
+`linalg` and `signal` have partial coverage in progress, and some of the compiled-code functions in `special`
+have been covered, with additional calling-out to CuPy, PyTorch and JAX, in
+[the `special` PR](https://github.com/scipy/scipy/pull/19023/), as mentioned above.
+[The tracker issue](https://github.com/scipy/scipy/issues/18867)
+should be kept up to date with progress as more submodules are converted.
+
+We currently support CuPy and PyTorch through `array-api-compat`, with Dask and JAX on the horizon.
+The great thing is that whenever a provider library complies with the standard, it will work with our
+converted submodules, without SciPy even needing to know that it exists!
+
+## Looking to the Future
+
+I have opened [the first PR for `scipy.linalg`](https://github.com/scipy/scipy/pull/19260)
+which covers the functions which are part of
+[the `linalg` standard extension module](https://data-apis.org/array-api/latest/extensions/linear_algebra_functions.html).
+In the short term, we hope to see PRs covering the rest of the submodules and continued refinement of our
+testing infrastructure.
+Looking further ahead, once all libraries which we want to support comply with the standard, we can drop
+`array-api-compat`.
```

Comment:
`Looking further ahead` to me make a stronger connection with the previous statement. i.e. still linked to SciPy although the rest of the sentence talks about the ecosystem in general. I would either rephrase the beginning of the sentence or make 2 different statements. 1 for the long term goal of SciPy (is there more to do after converting the modules) and 2 the long term goal for the whole Scientific Python ecosystem.